src/backend/functions/change-requests.ts

```diff
@@ -110,12 +111,102 @@ const getChangeRequestByID: ApiRouteFunction = async (params: { id: string }) =>
   return buildSuccessResponse(changeRequestTransformer(requestedCR));
 };
 
+const createNewChangeRequest: ApiRouteFunction = async (_params, event) => {
+  if (event.body === undefined) {
```

Comment:
A majority of these if statements serve to validate the data that we want. I'm going to provide two solutions below that will help make this code more readable. Logically all of the code makes sense to me, so it gets my approval. I did not manually run the code, but for your PR specifically, I assume you checked that. I do not know why each field needs to be validated, so I'll leave that to Alexander.

My first suggestion is adding a validation library. Feel free to google "validating data for typescript", "data validation javascript" etc. The main thing we want to do is validate data and there are libraries to do that.

Essentially these libraries let us define schemas for the data we want. Within these schemas we can say some fields are non-null or this must be a boolean or this must be an integer between 1-10 or define enums.

Here's a quick blog post to skim: https://blog.logrocket.com/dynamic-type-validation-in-typescript/

Reply:
I'm strongly in favor of evaluating a validation library to reduce the work we need to do, but it still takes time to understand even just one validation library to consider and I'm very much a back-end novice here. I agree Middy.js seems like a great choice, but one hesitation is that it seems clear to me from [their GitHub](https://github.com/middyjs/middy/issues/589) and from my test implementation (#414) that there isn't great TypeScript support. I don't currently think that's a deal-breaker though, but I haven't really evaluated any other packages.